src/app/GitUI/UserControls/FileStatusList.StatusSorter.cs

```diff
@@ -0,0 +1,146 @@
+﻿#nullable enable
+
+using GitExtensions.Extensibility.Git;
+using GitUI.UserControls;
+
+namespace GitUI;
+
+partial class FileStatusList
+{
+    internal sealed class StatusSorter : IStatusSorter
+    {
+        public TreeNode CreateTreeSortedByPath(IEnumerable<GitItemStatus> statuses, bool flat, Func<GitItemStatus, TreeNode> createNode)
+        {
+            TreeNode root = CreateParent(RelativePath.From(""));
+
+            TreeNode parent = root;
+            foreach (GitItemStatus status in statuses.OrderBy(s => s, new PathFirstComparer()))
+            {
+                parent = flat ? root : GetOrCreateParent(parent, status.Path, root);
+                TreeNode leaf = createNode(status);
+                parent.Nodes.Add(leaf);
+            }
+
+            root.Items().ForEach(RemoveParentPath);
+
+            return root;
+
+            static void RemoveParentPath(TreeNode node)
+            {
+                if (node.Parent?.Tag is RelativePath parentPath)
+                {
+                    if (parentPath.Length > 0 && node.Text.StartsWith(parentPath.Value))
+                    {
+                        node.Text = node.Text[(parentPath.Length + 1)..];
+                    }
+                }
+            }
+        }
+
+        private static TreeNode CreateParent(RelativePath relativePath) => new(text: relativePath.Value) { Tag = relativePath };
+
+        private static RelativePath GetCommonPath(RelativePath relativePathA, RelativePath relativePathB)
+        {
+            string a = $"{relativePathA}/";
+            string b = $"{relativePathB}/";
+            for (int commonEnd = 0; ; ++commonEnd)
+            {
+                if (commonEnd >= a.Length || commonEnd >= b.Length || a[commonEnd] != b[commonEnd])
+                {
+                    // Revert possible partial match
+                    while (commonEnd > 0 && a[--commonEnd] != '/')
+                    {
+                    }
+
+                    return RelativePath.From(a[..commonEnd]);
+                }
+            }
+        }
+
+        private static TreeNode GetOrCreateParent(TreeNode previousParent, RelativePath currentPath, TreeNode root)
+        {
+            RelativePath previousPath = (RelativePath)previousParent.Tag;
+            if (previousPath == currentPath)
```

Comment:
compile error in .net8
    C:\dev\gc\gitextensions\src\app\GitUI\UserControls\FileStatusList.StatusSorter.cs(62,41): error CS8600: Converting null literal or possible null value to non-nullable type.
    C:\dev\gc\gitextensions\src\app\GitUI\UserControls\FileStatusList.StatusSorter.cs(68,53): error CS8604: Possible null reference argument for parameter 'relativePathA' in 'RelativePath StatusSorter.GetCommonPath(RelativePath relativePathA, RelativePath relativePathB)'.

```suggestion
            if (previousParent.Tag is not RelativePath previousPath || previousPath == currentPath)
```

Reply:
It compiles for me with 8 SDK and 8.0.0 requested as well as with 9 SDK and 8.0 requested, and also on AppVeyor.

`Tag` must contain a `RelativePath` here. Though I must admit that `Tag` itself is not checked for `null`. All nodes have a Tag.
It shall crash otherwise.
Changing and force-pushing.